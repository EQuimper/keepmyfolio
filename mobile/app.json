{
  "expo": {
    "name": "keepmyfolio",
    "description": "An empty new project",
    "slug": "keepmyfolio",
    "privacy": "public",
    "sdkVersion": "20.0.0",
    "version": "1.0.0",
    "orientation": "portrait",
    "primaryColor": "#cccccc",
    "icon": "./assets/icons/app-icon.png",
    "loading": {
      "icon": "./assets/icons/loading-icon.png",
      "hideExponentText": false
    },
    "packagerOpts": {
      "assetExts": ["ttf", "mp4"]
    },
    "ios": {
      "supportsTablet": true
    }
  }
}
