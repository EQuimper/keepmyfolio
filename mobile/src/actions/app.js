// @flow

import type { Action } from '../types';

export function toggleTheme(): Action {
  return {
    type: 'app/TOGGLE_THEME',
  };
}

export function toggleSearchBar(): Action {
  return {
    type: 'app/TOGGLE_SEARCH_BAR',
  };
}
