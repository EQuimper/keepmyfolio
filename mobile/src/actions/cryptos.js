// @flow

import type { HoldingData } from '../types';

export function addNewHolding(coin: HoldingData) {
  return {
    type: 'cryptos/ADD_NEW_HOLDING',
    coin
  }
}
