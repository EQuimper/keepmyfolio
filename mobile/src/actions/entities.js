// @flow

import { CoinMarket } from '../utils/api';

import type { CoinMarketCapData } from '../types';

function getAllCoinMarketSuccess(data: CoinMarketCapData) {
  return {
    type: 'GET_ALL_COIN_MARKET_SUCCESS',
    data
  }
}

function getAllCoinMarketError(error: Error) {
  return {
    type: 'GET_ALL_COIN_MARKET_ERROR',
    error
  }
}

export function getAllCoinMarket() {
  return async (dispatch) => {
    dispatch({ type: 'GET_ALL_COIN_MARKET' });
    try {
      const { data } = await CoinMarket.getAll(10);
      return dispatch(getAllCoinMarketSuccess(data));
    } catch (error) {
      return dispatch(getAllCoinMarketError(error));
    }
  }
}
