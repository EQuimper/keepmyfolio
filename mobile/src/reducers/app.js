// @flow

import type { AppState, Action } from '../types';

import { themes } from '../utils/constants';

const initialState: AppState = {
  darkTheme: true,
  theme: themes.dark,
  isSearchBarShow: false,
};

export default function appReducer(
  state: AppState = initialState,
  action: Action,
) {
  switch (action.type) {
    case 'app/TOGGLE_THEME':
      return {
        ...state,
        darkTheme: !state.darkTheme,
        theme: themes[state.darkTheme ? 'light' : 'dark'],
      };
    case 'app/TOGGLE_SEARCH_BAR':
      return {
        ...state,
        isSearchBarShow: !state.isSearchBarShow,
      };
    default:
      return state;
  }
}
