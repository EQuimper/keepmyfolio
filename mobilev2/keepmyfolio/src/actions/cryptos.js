// @flow

import type { HoldingData, Action } from '../types';

export function addNewHolding(coin: HoldingData): Action {
  return {
    type: 'cryptos/ADD_NEW_HOLDING',
    coin
  }
}
