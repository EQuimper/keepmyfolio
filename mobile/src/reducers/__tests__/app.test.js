// @noflow

import reducer, { StateRecord } from '../app';
import { toggleSearchBar, toggleTheme } from '../../actions/app';

describe('App Reducer', () => {
  describe('Initial State', () => {
    it('should return the initialState on default', () => {
      const state = new StateRecord();

      expect(reducer(state, { type: 'INITIAL' })).toMatchSnapshot();
    });
  })

  describe('#toggleSearchBar()', () => {
    it('should toggle the searchBar to true if the state was false', () => {
      const state = new StateRecord();

      expect(reducer(state, toggleSearchBar())).toMatchSnapshot();
    });

    it('should toggle the searchBar to false if the state was true', () => {
      let state = new StateRecord();

      state = reducer(state, toggleSearchBar());
      state = reducer(state, toggleSearchBar());

      expect(state).toMatchSnapshot();
    });
  });

  describe('#toggleTheme()', () => {
    it('should put darkTheme to false if was true before and theme to light one', () => {
      const state = new StateRecord();

      expect(reducer(state, toggleTheme())).toMatchSnapshot();
    });

    it('should put darkTheme to true if was false before and theme to dark one', () => {
      let state = new StateRecord();

      state = reducer(state, toggleTheme());
      state = reducer(state, toggleTheme());

      expect(state).toMatchSnapshot();
    });
  });
});
