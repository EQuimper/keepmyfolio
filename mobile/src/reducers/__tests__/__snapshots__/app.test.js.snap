// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Reducer #toggleSearchBar() should toggle the searchBar to false if the state was true 1`] = `
Immutable.Record {
  "darkTheme": true,
  "theme": Object {
    "black": "#000",
    "cardBackground": "#161C36",
    "darkBlue": "#161C36",
    "darkGray": "#31393C",
    "green": "#4ADF86",
    "greyOutline": "#bbbbbb",
    "headerTitleColor": "#ffffff",
    "lightGrey": "#B4B4B4",
    "primary": "#00C0FF",
    "primaryLight": "rgba(0, 192, 255, 0.15)",
    "red": "#F8333C",
    "selectionColor": "#00C0FF",
    "tabBarColor": "#192543",
    "textColor": "#ffffff",
    "thumbTintColor": "#ffffff",
    "transparent": "transparent",
    "white": "#ffffff",
  },
  "isSearchBarShow": false,
}
`;

exports[`App Reducer #toggleSearchBar() should toggle the searchBar to true if the state was false 1`] = `
Immutable.Record {
  "darkTheme": true,
  "theme": Object {
    "black": "#000",
    "cardBackground": "#161C36",
    "darkBlue": "#161C36",
    "darkGray": "#31393C",
    "green": "#4ADF86",
    "greyOutline": "#bbbbbb",
    "headerTitleColor": "#ffffff",
    "lightGrey": "#B4B4B4",
    "primary": "#00C0FF",
    "primaryLight": "rgba(0, 192, 255, 0.15)",
    "red": "#F8333C",
    "selectionColor": "#00C0FF",
    "tabBarColor": "#192543",
    "textColor": "#ffffff",
    "thumbTintColor": "#ffffff",
    "transparent": "transparent",
    "white": "#ffffff",
  },
  "isSearchBarShow": true,
}
`;

exports[`App Reducer #toggleTheme() should put darkTheme to false if was true before and theme to light one 1`] = `
Immutable.Record {
  "darkTheme": false,
  "theme": Object {
    "black": "#000",
    "cardBackground": "#ffffff",
    "darkBlue": "#161C36",
    "darkGray": "#31393C",
    "green": "#4ADF86",
    "greyOutline": "#bbbbbb",
    "headerTitleColor": "#31393C",
    "lightGrey": "#B4B4B4",
    "primary": "#00C0FF",
    "primaryLight": "rgba(0, 192, 255, 0.15)",
    "red": "#F8333C",
    "selectionColor": "#00C0FF",
    "tabBarColor": "#F3F3F3",
    "textColor": "#31393C",
    "thumbTintColor": "#31393C",
    "transparent": "transparent",
    "white": "#ffffff",
  },
  "isSearchBarShow": false,
}
`;

exports[`App Reducer #toggleTheme() should put darkTheme to true if was false before and theme to dark one 1`] = `
Immutable.Record {
  "darkTheme": true,
  "theme": Object {
    "black": "#000",
    "cardBackground": "#161C36",
    "darkBlue": "#161C36",
    "darkGray": "#31393C",
    "green": "#4ADF86",
    "greyOutline": "#bbbbbb",
    "headerTitleColor": "#ffffff",
    "lightGrey": "#B4B4B4",
    "primary": "#00C0FF",
    "primaryLight": "rgba(0, 192, 255, 0.15)",
    "red": "#F8333C",
    "selectionColor": "#00C0FF",
    "tabBarColor": "#192543",
    "textColor": "#ffffff",
    "thumbTintColor": "#ffffff",
    "transparent": "transparent",
    "white": "#ffffff",
  },
  "isSearchBarShow": false,
}
`;

exports[`App Reducer Initial State should return the initialState on default 1`] = `
Immutable.Record {
  "darkTheme": true,
  "theme": Object {
    "black": "#000",
    "cardBackground": "#161C36",
    "darkBlue": "#161C36",
    "darkGray": "#31393C",
    "green": "#4ADF86",
    "greyOutline": "#bbbbbb",
    "headerTitleColor": "#ffffff",
    "lightGrey": "#B4B4B4",
    "primary": "#00C0FF",
    "primaryLight": "rgba(0, 192, 255, 0.15)",
    "red": "#F8333C",
    "selectionColor": "#00C0FF",
    "tabBarColor": "#192543",
    "textColor": "#ffffff",
    "thumbTintColor": "#ffffff",
    "transparent": "transparent",
    "white": "#ffffff",
  },
  "isSearchBarShow": false,
}
`;
